pkg/autogen/v2/autogen.go

```diff
@@ -301,3 +301,46 @@ func (a *ImplAutogenV2) ExtractPodSpec(resource unstructured.Unstructured) (*uns
 
 	return &unstructured.Unstructured{Object: podSpec}, nil
 }
+
+type ExtractPodFunc func(resource *unstructured.Unstructured) (*unstructured.Unstructured, error)
+
+func GetPodExtractor(kind string) ExtractPodFunc {
+	return func(resource *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+		autogen := &ImplAutogenV2{}
+		return autogen.ExtractPodSpec(*resource)
```

Comment:
you dont have to create a new struct here, add GetPodExtractor as a method to the struct, since it is not used by any other package, it should be a private func